docs/api_auth.md

```diff
@@ -42,6 +42,37 @@ sequenceDiagram
     Relay->>Extension: [{id:, address:, etc.}, {id:, address:, etc.}]
 ```
 
+## Firefox OAuth Token Authentication and Accept Terms of Service
+
+Similarly to the add-on, Firefox uses the [the FXA OAuth service][fxa-oauth] /oauth/token endpoint with `scope: ["https://identity.mozilla.com/apps/relay"]` to get the scoped access token that expires every 24 hours (see which calls [`getOAuthToken`][searchfox-getoauthtoken] [`accessTokenWithSessionToken`][searchfox-accesstokenwithsessiontoken]).
+
+After successful OAuth flow with FXA, Firefox uses this token to authenticate all requests to Relay by including an `Authorization: Bearer {fxa-access-token}` header in all API requests. Since the token was generated from FXA and not in Relay, Firefox must hit Relay's /api/v1/terms-accepted-user endpoint to ensure that user has accepted the Terms of Service and for a new user and profile to be created on Relay.
+
+```mermaid
+sequenceDiagram
+    participant Firefox
+    participant FXA
+    participant Relay
+
+rect rgba(34, 0, 51, .09)
+    note right of Firefox: OAuth2 Flow
+    Firefox->>FXA: POST /oauth/token/
+    FXA->>Firefox: JSON: {Relay access token valid for 24 hrs}
+    end
+    Firefox->>Relay: POST /api/v1/terms-accepted-user Authorization: Token {token}
+    Relay->>FXA: POST /v1/introspect {token}
+    FXA->>Relay: 200 OK
+    Relay->>Relay: Create new user and profile on DB if FxA user does not exist
+    Relay->>Firefox: 201 Created or 202 User already exists
+    note right of Firefox: Request new Relay Address
+    Firefox->>Relay: POST /api/v1/relayaddresses Authorization: Token {token}
+    Note over Relay,FXA: FxaTokenAuthentication
```

Comment:
suggestion (non-blocking): same as above - we could use `loop` or `alt` syntax here instead of `Note`.

Reply:
Completed: c4b8e2c